apps/fabric-website/src/components/Nav/Nav.tsx

```diff
@@ -38,32 +47,113 @@ export class Nav extends React.Component<INavProps, INavState> {
     );
   }
 
+  private _renderLinkList(pages: INavPage[], isSubMenu: boolean): React.ReactElement<{}> {
+    const { filterState } = this.state;
+
+    const links = pages
+      .filter(page => !page.hasOwnProperty('isHiddenFromMainNav'))
+      .map((page: INavPage, linkIndex: number) => {
+        if (page.isCategory && !filterState) {
+          return (
+            <span>
+              {page.pages.map((innerPage: INavPage, innerLinkIndex) => this._renderLink(innerPage, innerLinkIndex))}
+            </span>
+          );
+        }
+        return page.isCategory && filterState
+          ? this._renderCategory(page, linkIndex)
+          : this._renderLink(page, linkIndex);
+      });
+
+    return (
+      <ul className={css(styles.links, isSubMenu ? styles.isSubMenu : '')} aria-label="Main website navigation">
+        {links}
+      </ul>
+    );
+  }
+
+  private _renderCategory(page: INavPage, categoryIndex: number): React.ReactElement<{}> {
+    if (page.isCategory && page.pages) {
+      return (
+        <span key={categoryIndex} className={css(styles.category, _hasActiveChild(page) && styles.hasActiveChild)}>
+          <CollapsibleSection
+            titleAs={CollapsibleSectionTitle}
+            titleProps={{ text: page.title }}
+            defaultCollapsed={!_hasActiveChild(page)}
+          >
+            {page.pages.map((innerPage: INavPage, indexNumber: number) => this._renderLink(innerPage, indexNumber))}
+          </CollapsibleSection>
+        </span>
+      );
+    }
+  }
+
+  private _renderSortedLinks(pages: INavPage[]): React.ReactElement<{}> {
+    const links: INavPage[] = [];
+    pages.map((page: INavPage) => page.pages.map((link: INavPage) => links.push(link)));
+    links.sort((l1, l2) => {
+      if (l1.title > l2.title) {
+        return 1;
+      } else if (l1.title < l2.title) {
+        return -1;
+      }
+      return 0;
+    });
+
+    return this._renderLinkList(links, true);
+  }
+
   private _renderLink(page: INavPage, linkIndex: number): React.ReactElement<{}> {
     const ariaLabel = page.pages ? 'Hit enter to open sub menu, tab to access sub menu items.' : '';
     const title = page.title === 'Fabric' ? 'Home page' : page.title;
-    const childLinks = page.pages ? this._renderLinkList(page.pages, true) : null;
+    const childLinks =
+      page.pages && title === 'Components' && !this.state.filterState
+        ? this._renderSortedLinks(page.pages)
+        : page.pages
+          ? this._renderLinkList(page.pages, true)
+          : null;
+    const { searchQuery } = this.state;
+    const searchRegEx = new RegExp(searchQuery, 'i');
```

Comment:
Regexp is cleaner and matches empty searches so it cuts down on conditional checks ( `if (this.state.searchQuery.length > 0 && substring === to this.state.searchQuery blah blah) {...}` ) and reduces the need for `toLowerCase()` on both the link titles and the search queries.

Reply:
I just changed the line of the render link and instead of using the regex I added this: (page.isFilterable && searchQuery !== '' ? matchIndex > -1 : true) but on the part where the search is highlighted I took out the match index so that I could use it there